content/en/docs/tasks/manage-secrets/creating-your-own-secrets.md

```diff
@@ -0,0 +1,288 @@
+  username: administrator
+```
+
+Results in the following secret:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  creationTimestamp: 2018-11-15T20:46:46Z
+  name: mysecret
+  namespace: default
+  resourceVersion: "7579"
+  selfLink: /api/v1/namespaces/default/secrets/mysecret
+  uid: 91460ecb-e917-11e8-98f2-025000000001
+type: Opaque
+data:
+  username: YWRtaW5pc3RyYXRvcg==
+```
+
+Where `YWRtaW5pc3RyYXRvcg==` decodes to `administrator`.
+
+The keys of data and stringData must consist of alphanumeric characters,
+'-', '_' or '.'.
+
+**Encoding Note:** The serialized JSON and YAML values of secret data are
+encoded as base64 strings.  Newlines are not valid within these strings and must
+be omitted.  When using the `base64` utility on Darwin/macOS users should avoid
+using the `-b` option to split long lines.  Conversely Linux users *should* add
+the option `-w 0` to `base64` commands or the pipeline `base64 | tr -d '\n'` if
+`-w` option is not available.
+
+## Creating a Secret from Generator
+Kubectl supports [managing objects using Kustomize](/docs/tasks/manage-kubernetes-objects/kustomization/)
+since 1.14. With this new feature,
+you can also create a Secret from generators and then apply it to create the object on
+the Apiserver. The generators
+should be specified in a `kustomization.yaml` inside a directory.
+
+For example, to generate a Secret from files `./username.txt` and `./password.txt`
+```shell
+# Create a kustomization.yaml file with SecretGenerator
+cat <<EOF >./kustomization.yaml
+secretGenerator:
+- name: db-user-pass
+  files:
+  - username.txt
+  - password.txt
+EOF
+```
+Apply the kustomization directory to create the Secret object.
+```shell
+$ kubectl apply -k .
+secret/db-user-pass-96mffmfh4k created
+```
+
+You can check that the secret was created like this:
+
+```shell
+$ kubectl get secrets
+NAME                             TYPE                                  DATA      AGE
+db-user-pass-96mffmfh4k          Opaque                                2         51s
+
+$ kubectl describe secrets/db-user-pass-96mffmfh4k
+Name:            db-user-pass
+Namespace:       default
+Labels:          <none>
+Annotations:     <none>
+
+Type:            Opaque
+
+Data
+====
+password.txt:    12 bytes
+username.txt:    5 bytes
+```
+
+For example, to generate a Secret from literals `username=admin` and `password=secret`,
+you can specify the secret generator in `kustomization.yaml` as
+```shell
+# Create a kustomization.yaml file with SecretGenerator
+$ cat <<EOF >./kustomization.yaml
+secretGenerator:
+- name: db-user-pass
+  literals:
+  - username=admin
+  - password=secret
+EOF
+```
+Apply the kustomization directory to create the Secret object.
+```shell
+$ kubectl apply -k .
+secret/db-user-pass-dddghtt9b5 created
+```
+{{< note >}}
+The generated Secrets name has a suffix appended by hashing the contents. This ensures that a new
+Secret is generated each time the contents is modified.
+{{< /note >}}
+
+{{% /capture %}}
```

Comment:
Does this page warrant a What's Next section?

Reply:
@sftim I have added What's Next section above. What do you think about having "Create a Pod that has access to the secret data through a Volume" as next reading suggestion?

Reply:
Sounds good to me. Also consider https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/

Reply:
Actually, I wonder whether it makes sense to snip that content out of the Concept page and merge it into https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/

Otherwise, there will be two Tasks that cover similar ground. I had not realised that when I logged the original issue.

Reply:
Do you mean taking out https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets (Using Secrets) section from concept -> Secrets page and merge into https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/ ?

There is also duplication of https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#create-a-secret with our Tasks page "Creating Your Own Secrets" (https://deploy-preview-16176--kubernetes-io-master-staging.netlify.com/docs/tasks/manage-secrets/creating-your-own-secrets/ ).

Reply:
There's different ways of tidying this up. What I've realized here is that my original recommendations don't necessarily work well. Sorry about that.